tests/python/sparsetir/sparse_tir_lowered_buffer_scripts.py

```diff
@@ -101,7 +101,7 @@ def csrmm_dense_iter(
             low = T.alloc_buffer([1], dtype="int32", strides=[1], scope="local")
             high = T.alloc_buffer([1], dtype="int32", strides=[1], scope="local")
             low[0] = 0
-            high[0] = J_indptr_data[vi + 1] - J_indptr_data[vi]
+            high[0] = J_indptr_data[vi + 1] - J_indptr_data[vi] - 1
```

Comment:
https://github.com/uwsampl/SparseTIR/blob/ca59cbe4e81de959b798f4e5cdd9c5ab6ea7f801/src/tir/transforms/lower_sparse_iter.cc#L828-L829
In the function you set the upper bound as `indptr[i+1]-indptr[i]`
from these line
https://github.com/uwsampl/SparseTIR/blob/ca59cbe4e81de959b798f4e5cdd9c5ab6ea7f801/src/tir/transforms/lower_sparse_iter.cc#L923-L925
we can see if the val is greater than indices[i,indptr[i+1]-indptr[i]-1],it's possable that the mid buffer we return contains the value of indices[i,indptr[i+1]-indptr[i]],which is wrong.

Reply:
What's more,if the val of index just equal to `indices[i,indptr[i+1]-indptr[i]]` then it will return a valid index,and cause the wrong result.

Reply:
OK，I can fix this later

Reply:
It looks like you choose the same way (by adding a judgment condition) to fix this problem like me in my newest commit
```
Stmt then_stmt = BufferStore(mid, -1, mid_indices);
PrimExpr if_stmt = (pivot != val || mid_val == ub);
body_stmts.push_back(IfThenElse(if_stmt, then_stmt));
```
However,I don't know why you return a pair in `binary_search` and it seems the second item is not used in your commit.Do you mean leave the valid check outside the binary search function?

Reply:
Sorry I didn't notice your commit.
The valid flag is also returned to tell the caller of `BinarySearch` function whether the search is successful, so that we don't need to access `buf[..., mid[...]]` outside the binary search body.

We don't need exact value because the valid flag indicate such information, if the search is successful then the value equals `buf[..., ,mid[...]]`.

Reply:
In your design you use a buffer `success` to record the valid flag of every element of `mid`,but if this 

> Hi @qelk123 do you need any assistance on rebase? It seems the codebase disappears and the PR is closed.



> Hi @qelk123 do you need any assistance on rebase? It seems the codebase disappears and the PR is closed.

It's OK ,I will open a new pr later.